bugs/lp-1889555.txt

```diff
@@ -0,0 +1,1210 @@
+=== Begin SRU Template ===
+[Impact]
+Older versions of cloud-init could misconfigure grub on nvme devices,
+which could prevent instances from booting after a grub upgrade.
+
+[Test Case]
+For focal, bionic, and xenial verify the following:
+1. on an affected instance, test that installing the new version of cloud-init appropriately updates debconf
+2. on an affected instance, modify of the debconf settings and test that installing the new version of cloud-init does not touch those values
+3. in a container, confirm that cloud-init does not touch the values
+4. in an unaffected instance (i.e. one without an NVMe root), confirm that cloud-init does not touch the values
+
+Steps for test 1:
+# Find an old affected image with
+aws ec2 describe-images --filters "Name=name,Values=Ubuntu <release number>*"
+
+# Launch an AWS with affected image-id, ssh in
+
+# After startup, connect via SSH, then
+# Verify we're on an nvme device
+lsblk | grep nvme
+
+# Verify install_devices set incorrectly
+debconf-show grub-pc | grep "install_devices:"
+
+# update cloud-init to proposed
+mirror=http://archive.ubuntu.com/ubuntu
+echo deb $mirror $(lsb_release -sc)-proposed main | tee /etc/apt/sources.list.d/proposed.list
+apt-get update -q
+apt-get install -qy cloud-init
+
+# Verify "Reconfiguring grub" message in upgrade output
+
+# Verify install_devices set correctly
+debconf-show grub-pc | grep "install_devices:"
+
+# Verify that after reboot we can still connect
+
+Steps for test 2:
+# Find an old affected image with
+aws ec2 describe-images --filters "Name=name,Values=Ubuntu <release number>*"
+
+# Launch an AWS with affected image-id
+
+# After startup, connect via SSH, then
+# Verify we're on an nvme device
+lsblk | grep nvme
+
+# Verify install_devices set incorrectly
+debconf-show grub-pc | grep "install_devices:"
+
+# Update install device to something (anything) else
+echo 'set grub-pc/install_devices /dev/sdb' | debconf-communicate
+
+# update cloud-init to proposed
+mirror=http://archive.ubuntu.com/ubuntu
+echo deb $mirror $(lsb_release -sc)-proposed main | tee /etc/apt/sources.list.d/proposed.list
+apt-get update -q
+apt-get install -qy cloud-init
+
+# Verify no "Reconfiguring grub" message in upgrade output
+# Verify install_devices not changed
+debconf-show grub-pc | grep "install_devices:"
+
+Steps for test 3:
+# lxd launch affected image
+lxc launch <image>
+
+# Obtain bash shell
+lxc exec <image> bash
```

Comment:
```suggestion
lxc exec <instance> bash
```

And for future reference (though as you've already done validation with `exec` we shouldn't change it here), you can get a shell with:

```suggestion
lxc shell <instance>
```